src/lavinmq/amqp/exchange/topic.cr

```diff
@@ -51,74 +155,17 @@ module LavinMQ
         # optimize the case where the only binding key is '#'
         if bindings.size == 1
           bk, destinations = bindings.first
-          if bk.size == 1
-            if bk.first == "#"
-              destinations.each do |destination, _binding_key|
-                yield destination
-              end
+          if bk.acts_as_fanout?
+            destinations.each do |destination, _binding_key|
+              yield destination
             end
+            return
           end
         end
 
-        rk_parts = routing_key.split(".")
+        rk = RkIterator.new(routing_key.to_slice)
```

Comment:
@spuun need some 👀 on this. All specs passes but creating `rk` an instance of `RkIterator` would that be consumed on the first loop of bindings and then fail the next iteration? It doesn't, is that because it's a struct and we copy it into each `matches?` call?